src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.Reflection.cs

```diff
@@ -22,10 +21,6 @@ internal partial class XamlFileGenerator
 		private Func<INamedTypeSymbol, string[]>? _findLocalizableDeclaredProperties;
 		private XClassName? _xClassName;
 		private string[]? _clrNamespaces;
-		private readonly static Func<INamedTypeSymbol, IPropertySymbol?> _findContentProperty;
-		private readonly static Func<INamedTypeSymbol, string, bool> _isAttachedProperty;
-		private readonly static Func<INamedTypeSymbol, string, INamedTypeSymbol> _getAttachedPropertyType;
-		private readonly static Func<INamedTypeSymbol, bool> _isTypeImplemented;
```

Comment:
Thinking again about this PR. We need to reason about what to memoize and what not to.

In all cases, the memoized function should not be static as this will leak.

It may be worth to still keep some/all of these memoized, but non-static.

I guess it's safer to keep the memoization for all of them for now?

In this case, these should be moved to XamlCodeGeneration or RoslynMetadataHelper rather than XamlFileGenerator so that all files share the same cache.

Reply:
Draft for now until I address this comment.